### PR TITLE
Feature/filestorage securityservices create

### PR DESCRIPTION
**Pull request by ehdou**
Add support for Security Service Create
For #131

### CODE REVIEW

**Comment by ehdou** [comment]
Here is the [DB](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/db/sqlalchemy/models.py#L728) representation of security services, the [view](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/views/security_service.py) and the [create function](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v1/security_service.py#L191)

**Comment by ehdou** [comment]
@jrperritt This PR is ready for review. It's based on [PR#126](https://github.com/gophercloud/gophercloud/pull/126) and adds support for security service Create

**Comment by ehdou** [comment]
@jrperritt This PR is ready for review

File: openstack/sharedfilesystems/v2/securityservices/requests.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+// CreateOpts contains options for creating a SecurityService. This object is
+// passed to the securityservices.Create function. For more information about
+// these parameters, see the SecurityService object.
+type CreateOpts struct {
+	// The security service name
+	Name string `json:"name"`
+	// The security service description
+	Description string `json:"description,omitempty"`
+	// The security service type. A valid value is ldap, kerberos, or active_directory
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/securityservices/requests.go]: Are these the only [3] types allowed? If so, we should probably create a type for this that has those three as `const`s of that type.

File: openstack/sharedfilesystems/v2/securityservices/requests.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+type CreateOptsBuilder interface {
+	ToSecurityServiceCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts contains options for creating a SecurityService. This object is
+// passed to the securityservices.Create function. For more information about
+// these parameters, see the SecurityService object.
+type CreateOpts struct {
+	// The security service name
+	Name string `json:"name"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/securityservices/requests.go]: Can the name by empty or should this have an `omitempty` or `required:true` tag?

File: openstack/sharedfilesystems/v2/securityservices/requests.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	// The security service description
+	Description string `json:"description,omitempty"`
+	// The security service type. A valid value is ldap, kerberos, or active_directory
+	Type string `json:"type" required:"true"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/securityservices/requests.go]: Let's move required fields to the top of the struct.

File: openstack/sharedfilesystems/v2/securityservices/requests.go (hunk 2 above)
Comment (ehdou) [inline openstack/sharedfilesystems/v2/securityservices/requests.go]: The name is optional, I added `omitempty`

File: openstack/sharedfilesystems/v2/securityservices/requests.go (hunk 1 above)
Comment (ehdou) [inline openstack/sharedfilesystems/v2/securityservices/requests.go]: good point, done

**Comment by ehdou** [comment]
@jrperritt This PR is ready for re-review

**Pull request merged**